packages/components/link/src/use-link.ts

```diff
@@ -44,8 +44,12 @@ export type UseLinkProps = Props & AriaLinkProps;
 
 export function useLink(originalProps: UseLinkProps) {
   const globalContext = useProviderContext();
+  const resolvedProps =
+    globalContext?.useHref && originalProps.href
+      ? {...originalProps, href: globalContext.useHref(originalProps.href)}
+      : originalProps;
```

Comment:
This logic seems to be covered in `useLinkProps` in `packages/hooks/use-aria-link/src/index.ts` already. Could you please cross check the react-spectrum implementation (`packages/@react-aria/link/src/useLink.ts`)? From what I see, they look same.